### PR TITLE
[tado] Add channel for remaining time of open window override

### LINKED ISSUE #17558: [tado] openWindowDetected not available in API response
Hi, I recently added an item to the "Open Window" channel in the tado° binding. But it keeps staying "OFF".

I checked the API response via the webapp/curl and it seems that the `openWindowDetected` property does not exist (anymore?). Only the `openWindow` is there.

This is the API response for `.../zones/[id]/state` when an open window is detected:

```
{"tadoMode":"HOME","geolocationOverride":false,"geolocationOverrideDisableTime":null,"preparation":null,"setting":{"type":"HEATING","power":"ON","temperature":{"celsius":20.00,"fahrenheit":68.00}},"overlayType":null,"overlay":null,"openWindow":{"detectedTime":"2024-10-13T12:52:17Z","durationInSeconds":1200,"expiry":"2024-10-13T13:12:17Z","remainingTimeInSeconds":16},"nextScheduleChange":{"start":"2024-10-13T19:40:00Z","setting":{"type":"HEATING","power":"ON","temperature":{"celsius":19.00,"fahrenheit":66.20}}},"nextTimeBlock":{"start":"2024-10-13T19:40:00.000Z"},"link":{"state":"ONLINE"},"runningOfflineSchedule":false,"activityDataPoints":{"heatingPower":{"type":"PERCENTAGE","percentage":0.00,"timestamp":"2024-10-13T12:56:08.950Z"}},"sensorDataPoints":{"insideTemperature":{"celsius":21.29,"fahrenheit":70.32,"timestamp":"2024-10-13T13:09:14.082Z","type":"TEMPERATURE","precision":{"celsius":0.1,"fahrenheit":0.1}},"humidity":{"type":"PERCENTAGE","percentage":70.50,"timestamp":"2024-10-13T13:09:14.082Z"}}}
```

And this if no open window is detected:

```
{"tadoMode":"HOME","geolocationOverride":false,"geolocationOverrideDisableTime":null,"preparation":null,"setting":{"type":"HEATING","power":"ON","temperature":{"celsius":20.50,"fahrenheit":68.90}},"overlayType":null,"overlay":null,"openWindow":null,"nextScheduleChange":{"start":"2024-10-13T17:30:00Z","setting":{"type":"HEATING","power":"ON","temperature":{"celsius":19.00,"fahrenheit":66.20}}},"nextTimeBlock":{"start":"2024-10-13T17:30:00.000Z"},"link":{"state":"ONLINE"},"runningOfflineSchedule":false,"activityDataPoints":{"heatingPower":{"type":"PERCENTAGE","percentage":0.00,"timestamp":"2024-10-13T13:10:24.641Z"}},"sensorDataPoints":{"insideTemperature":{"celsius":21.04,"fahrenheit":69.87,"timestamp":"2024-10-13T13:06:26.743Z","type":"TEMPERATURE","precision":{"celsius":0.1,"fahrenheit":0.1}},"humidity":{"type":"PERCENTAGE","percentage":74.80,"timestamp":"2024-10-13T13:06:26.743Z"}}}
```

Only `openWindow` is available in both cases.

Environment
* Version used: (e.g., openHAB and add-on versions): 4.2.1 release build
* Operating System and version (desktop or mobile, Windows 10, Raspbian Buster, ...): Openhabian Bookworm

Perhaps this needs to be corrected in the binding? Thanks!

### PR DESCRIPTION
**Pull request by andrewfg**
Resolves #17558 

Signed-off-by: AndrewFG <software@whitebear.ch>

### CODE REVIEW

**Comment by andrewfg** [comment]
@eengnr I have created this PR for you, but before we merge it it would be good if you can test and confirm that it functions as expected on your system.

The JAR file is in the zip file below. It will run on an OH v4.3 current milestone. You need to uninstall the stock binding, and drop the new Jar on your addons folder. And you may need to delete and recreate your Things so that the new channel will be created.

[org.openhab.binding.tado-4.3.0-SNAPSHOT.jar.zip](https://github.com/user-attachments/files/17396949/org.openhab.binding.tado-4.3.0-SNAPSHOT.jar.zip)

**Comment by eengnr** [comment]
> @eengnr I have created this PR for you, but before we merge it it would be good if you can test and confirm that it functions as expected on your system.
> 
> The JAR file is in the zip file below. It will run on an OH v4.3 current milestone. You need to uninstall the stock binding, and drop the new Jar on your addons folder. It should add the new channel automatically, but if not you may need to delete and recreate your Things so that the new channel will be created.
> 
> [org.openhab.binding.tado-4.3.0-SNAPSHOT.jar.zip](https://github.com/user-attachments/files/17397570/org.openhab.binding.tado-4.3.0-SNAPSHOT.jar.zip)

Thanks for the snapshot! The time is available and updated in the linked item every 30 seconds.

But the newly created switch item linked to the channel stays OFF. But that's not a problem imho, I can just create a rule to turn it ON/OFF based in the time.

**Comment by andrewfg** [comment]
> the newly created switch item linked to the channel stays OFF

Understood. But that is not how OH core should be working. Let me think about it.

**Comment by andrewfg** [comment]
@eengnr to properly link a **Switch** item to the `openWindowRemainingTime` channel, the item must be created with a **"Hysteresis" Profile** having a Lower Bound of <= 0 s (which maps to `OFF`) and an Upper Bound of >=1 s (which maps to `ON`)..

image

**Comment by eengnr** [comment]
@andrewfg Ah thanks! I'll give it a try.

The German translation for the explanation of the lower/upper bound is missing the "or equal", but I set 0/1 now as in your screenshot.

File: bundles/org.openhab.binding.tado/src/main/resources/OH-INF/thing/thing-types.xml [hunk 1]
```diff
@@ -307,4 +308,12 @@
 		<state readOnly="true"/>
 	</channel-type>
 
+	<channel-type id="openWindowRemainingTime">
+		<item-type>Number:Time</item-type>
+		<label>Open Window Override Remaining Time</label>
```
Comment (lsiepel) [inline bundles/org.openhab.binding.tado/src/main/resources/OH-INF/thing/thing-types.xml:313]: While i don't have a improvement, we need to have these labels 2-3 words and < 23 chars. 
With 5 words and 35 characters, this exceeds it with a serious amount. Hope you can think of something creative.

**Review by lsiepel — CHANGES_REQUESTED** [review]
Thanks, just one comment. Otherwise LGTM

File: bundles/org.openhab.binding.tado/src/main/resources/OH-INF/thing/thing-types.xml (hunk 1 above)
Comment (andrewfg) [reply bundles/org.openhab.binding.tado/src/main/resources/OH-INF/thing/thing-types.xml]: Ok. Done.

**Review by lsiepel — APPROVED** [review]
Thanks LGTM

**Pull request merged**